### DISCUSSION TITLE
C++11

### DISCUSSION THREAD

**Discussion by LoganDark** [taocpp/json#43] [Q&A]
Will the C++11 version be maintained any longer? Or are you deprecating it forever in favor of the newer standard?

**Comment by ColinH** [comment]
We are concentrating all efforts on the C++17 "version", the C++11 version is not deprecated, it's gone because it never really existed outside of betas and development versions: we made the switch to C++17 before releasing a version 1.0.0.

We will not continue with C++11, but if you are using a snapshot of taocpp/json with C++11 and can't upgrade to C++17 you can still ask questions and we will give "unofficial" support or help you back-port features where necessary and possible.

**Comment by LoganDark** [comment]
I'm a bit disappointed you decided to ditch C++11 at the last moment, because this library had the best performance of any JSON library, and it just so happens that C++11 is the most common version of C++ (that I'm aware of).

**Comment by ColinH** [comment]
Thanks for the praise! And sorry for the bother, there were simply too many simplifications and enhancements to the code made possible by using the newer standard, we couldn't resist any longer...

**Comment by d-frey** [comment]
...and by "couldn't resist" we also mean that we previously had to maintain backported versions of `std::optional` and `std::string_view`, which were not 100% compatible with the actual standardized version (as `std::string_view` also requires modifications to `std::string`, `constexpr` could not be used fully with C++11, etc.). It wasn't a light-hearted decision, it was more-or-less necessary and frees up more resources on our side.

**Comment by brlcad** [comment]
Ditto disappointment.  It knocks this outstanding implementation out of contention for our software stack, at least until 17 is reliably implemented and available everywhere, all platforms. :(
